Dockerfile.s3

```diff
@@ -0,0 +1,129 @@
+ARG TARGETARCH
+
+# set workdir
+RUN mkdir -p /opt/code/localstack
+WORKDIR /opt/code/localstack/
+
+# Install runtime OS package dependencies
+RUN --mount=type=cache,target=/var/cache/apt \
+    apt-get update && \
+        # Install dependencies to add additional repos
+        apt-get install -y --no-install-recommends \
+            # Runtime packages (groff-base is necessary for AWS CLI help)
+             ca-certificates curl make openssl
+# TODO: add this if we need the DNS server: iputils-ping iproute2
+
+SHELL [ "/bin/bash", "-c" ]
+
+# create localstack user and filesystem hierarchy, perform some permission fixes
+RUN chmod 777 . && \
+    useradd -ms /bin/bash localstack && \
+    mkdir -p /var/lib/localstack && \
+    chmod -R 777 /var/lib/localstack && \
+    mkdir -p /usr/lib/localstack && \
+    mkdir /tmp/localstack && \
+    chmod -R 777 /tmp/localstack && \
+    touch /tmp/localstack/.marker
+
+# install basic (global) tools to final image
+RUN --mount=type=cache,target=/root/.cache \
+    pip install --no-cache-dir --upgrade virtualenv
+
+# install the entrypoint script
+ADD bin/docker-entrypoint.sh /usr/local/bin/
+# add the shipped hosts file to prevent performance degredation in windows container mode on windows
+# (where hosts file is not mounted) See https://github.com/localstack/localstack/issues/5178
+ADD bin/hosts /etc/hosts
+
+# expose default environment
+# Set edge bind host so localstack can be reached by other containers
+# set library path and default LocalStack hostname
+ENV USER=localstack
+ENV PYTHONUNBUFFERED=1
+
+
+# builder: Stage which installs the dependencies of LocalStack Community
+FROM base as builder
+ARG TARGETARCH
+
+# Install build dependencies to base
+RUN --mount=type=cache,target=/var/cache/apt \
+    apt-get update && \
+        # Install dependencies to add additional repos
+        apt-get install -y gcc
+
+# upgrade python build tools
+RUN --mount=type=cache,target=/root/.cache \
+    (virtualenv .venv && . .venv/bin/activate && pip3 install --upgrade pip wheel setuptools)
+
+# add files necessary to install all dependencies
+ADD Makefile setup.py setup.cfg pyproject.toml ./
+# add the root package init to invalidate docker layers with version bumps
+ADD localstack/__init__.py localstack/
+# add the localstack start scripts (necessary for the installation of the runtime dependencies, i.e. `pip install -e .`)
+ADD bin/localstack bin/localstack.bat bin/localstack-supervisor bin/
+
+# install dependencies to run the LocalStack Pro runtime and save which ones were installed
+RUN --mount=type=cache,target=/root/.cache \
+    make install-s3
+RUN . .venv/bin/activate && pip3 freeze -l > requirements-runtime.txt
+
+# delete the botocore specs for other services (>80mb)
+# TODO: well now it's compressed and it's much lighter: 20mb maybe not worth it
+RUN find .venv/lib/python3.11/site-packages/botocore/data/ -mindepth 1 -maxdepth 1 -type d -not -name s3 -exec rm -rf '{}' \;
```

Comment:
I guess this is not great, but not terrible. If it doesn't cause any issues, we could leave this optimization in. But if it turns out that this could be problematic, I would vote for removing this (since it seems a bit hacky).

Reply:
I agree it's not great, but it saves a lot of space. It didn't create any issues yet, so I felt safe doing it, and I'm not seeing another way of doing this. Because if we don't do this, without the `SERVICES` variables, then this is basically a LocalStack image for all services fully implemented in LocalStack like SQS. Let's see then!